dotcom-rendering/src/web/components/SecureSignupIframe.importable.tsx

```diff
@@ -100,49 +100,58 @@ const postFormData = async (
 	});
 };
 
+type EventDescription =
+	| 'click-button'
+	| 'form-submission'
+	| 'submission-confirmed'
+	| 'submission-failed'
+	| 'open-captcha'
+	| 'captcha-load-error'
+	| 'form-submit-error'
+	| 'captcha-not-passed'
+	| 'captcha-passed';
+
 const sendTracking = (
 	newsletterId: string,
-	event:
-		| 'click-button'
-		| 'submit-form'
-		| 'form-submit-error'
-		| 'captcha-load-error'
-		| 'open-captcha'
-		| 'captcha-not-passed'
-		| 'captcha-passed'
-		| 'success-response-received'
-		| 'fail-response-received',
+	eventDescription: EventDescription,
 ): void => {
 	const ophanRecord = getOphanRecordFunction();
 
 	let action: OphanAction = 'CLICK';
-	const value = `${event} ${newsletterId}`;
 
-	switch (event) {
-		case 'submit-form':
-			action = 'SUBSCRIBE';
+	switch (eventDescription) {
+		case 'form-submission':
+		case 'captcha-not-passed':
+		case 'captcha-passed':
+			action = 'ANSWER';
 			break;
-		case 'open-captcha':
-			action = 'EXPAND';
+		case 'submission-confirmed':
+			action = 'SUBSCRIBE';
 			break;
-		case 'form-submit-error':
 		case 'captcha-load-error':
+		case 'form-submit-error':
+		case 'submission-failed':
 			action = 'CLOSE';
 			break;
-		case 'captcha-not-passed':
-		case 'captcha-passed':
-			action = 'ANSWER';
-			break;
-		case 'success-response-received':
-		case 'fail-response-received':
-			action = 'RETURN';
+		case 'open-captcha':
+			action = 'EXPAND';
 			break;
 		case 'click-button':
 		default:
 			action = 'CLICK';
 			break;
 	}
 
+	// The data team use a custom date format for timestamps,
+	// (yyy-MM-dd hh:mm:ss.ssssss UTC)
+	// and will cast the integer value  to this
+	// format at their end
+	const value = JSON.stringify({
+		eventDescription,
+		newsletterId,
+		timestamp: new Date().getDate(),
```

Comment:
@mxdvl made a good comment earlier around the problem with using timestamps on the client (they are surprisingly inaccurate) and made the point that server side time is preferable. But I think that really what we care about here is not _when_ an event happened but the _duration_ between two events, is that true?

If so, does it make more sense to capture this duration rather than a date?

Reply:
Yes, its the time elapsed between events that matters, but I don't see how we would build in a concept of duration between different events of several types without adding a lot of complexity and fragility to the component and increasing the amount of code we need users to download. Also, I'm not sure we could stretch the event model in Ophan to effectively communicate that information.

If you take a look at the "not signed in user" flowchart on this PR: https://github.com/guardian/frontend/pull/25158 , you'll see that the web of events is fairly complex (the chart is for the sign-up forms on the all newsletter's page, but it follows essentially the same logic).

Reply:
Ah, so it's not just the duration between event A and B but C,D and E as well? I agree in this case it would more complex to record durations vs. timestamps - thanks for the extra detail 👍